src/client/editor/components/ShadowComponent.js

```diff
@@ -4,18 +4,28 @@ import { types } from "./utils";
 export default class ShadowComponent extends BaseComponent {
   static componentName = "shadow";
 
+  static removable = false;
+
+  static dontExportProps = true;
+
   static schema = [
     { name: "castShadow", type: types.boolean, default: true },
     { name: "receiveShadow", type: types.boolean, default: true }
   ];
 
   updateProperty(propertyName, value) {
     super.updateProperty(propertyName, value);
-    this._object.traverse(obj => {
-      if (obj instanceof THREE.Mesh) {
-        obj[propertyName] = value;
-        obj.material.needsUpdate = true;
-      }
-    });
+    this._object[propertyName] = value;
+
+    if (this._object.material) {
+      this._object.material.needsUpdate = true;
+    }
```

Comment:
Hmm, I guess we should actually disallow putting this component on a non-Mesh object. 
Probably a task for after alpha.